addons/launcher/models/launcher.py

```diff
@@ -0,0 +1,47 @@
+"""
+Launcher is a model for starting threads (or possibly other things) when the
+server starts.
+
+Models can register functions to be run as threads by calling `register_method`
+from their `_register_hook` method.
+"""
+
+import logging
+import threading
+
+from odoo import models
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Launcher(models.TransientModel):
+    """A model that starts a thread for each of its records."""
+
+    _description = "Launches threads at server start time."
+    _name = "launcher.launcher"
+
+    _registered = []
+
+    @classmethod
+    def register_method(cls, method_obj):
+        """
+        Register methods to be called as threads.
+
+        Subclasses should call this method from their _register_hook method.
+        """
+        # Ideally we would use __init_subclass__ for this, but it doesn't seem
+        # to play well with Odoo's way of initialising subclasses.
+        cls._registered.append(method_obj)
+
+    def launch(self):
+        """Launches a thread for each registered callable."""
+        for i, launchable in enumerate(self._registered):
+            t = threading.Thread(target=launchable)
+            # FIXME?  Make it a daemon thread as it having difficulty finding a
```

Comment:
Is this FIXME still required?

Reply:
I'm in two minds about it.

While developing, I wanted to get it working such that the threads could be terminated, for example by passing an event object which they could inspect and end if it was set.  This might be useful if we wanted to clean up database connection etc when the threads exited.  However, the threads did not join even when the event was set, so I abandoned it and used daemon threads instead.

So the fixme partially explains why we are using daemon threads, and hints that a better approach might be welcome.

I'm happy to remove it now we've ben through review.

Reply:
Ah, I think I misunderstand what the FIXME was saying. I read it to mean you wanted to make it a daemon in future when you already had. 

Personally I'm not against a FIXME like this but maybe a re-wording and greater explanation

Reply:
I've revised the comment